src/handlers/storage-handler.ts

```diff
@@ -67,36 +72,78 @@ export class WrappedStorageHandler extends WrappedReadOnlyStorageHandler impleme
     return handler
   }
 
-  get(key: string): string | null {
-    if (strEqualsIgnoreCase(this.storageStrategy, StorageStrategies.none) || strEqualsIgnoreCase(this.storageStrategy, StorageStrategies.disabled)) {
+  private getCookieRecord(key: string): StorageRecord | null {
+    let expiresAt: Date | undefined
+
+    const cookieExpirationEntry = this.getCookie(expirationKey(key))
+    if (cookieExpirationEntry && cookieExpirationEntry.length > 0) {
+      expiresAt = new Date(cookieExpirationEntry)
+      if (expiresAt <= new Date()) {
+        return null
+      }
+    }
+
+    const data = this.getCookie(key)
+    if (data) {
+      return { data, expiresAt }
+    } else {
       return null
-    } else if (strEqualsIgnoreCase(this.storageStrategy, StorageStrategies.localStorage)) {
-      if (this.localStorageIsEnabled()) {
-        const expirationKey = `${key}_exp`
-        const oldLsExpirationEntry = this.getDataFromLocalStorage(expirationKey)
-        if (oldLsExpirationEntry && Date.parse(oldLsExpirationEntry) <= new Date().getTime()) {
+    }
+  }
+
+  private getLSRecord(key: string): StorageRecord | null {
+    if (this.localStorageIsEnabled()) {
+      let expiresAt: Date | undefined
+      const oldLsExpirationEntry = this.getDataFromLocalStorage(expirationKey(key))
+
+      if (oldLsExpirationEntry) {
+        expiresAt = new Date(oldLsExpirationEntry)
+        if (expiresAt <= new Date()) {
```

Comment:
Shall we handle the hopefully really rare edge case of not being able to parse the expiration date - e.g. by removing the entry?

Reply:
We can, though right now it will just return an invalid date as the expirationDate.
Would do you propose to do in such cases, not return any data at all?

Side note: None of liveconnect handles the case that something else writes to our cookies, so if we care about it we have more problematic locations to fix

Reply:
adjusted